src/Raven.Server/Documents/Indexes/Persistence/ConverterBase.cs

```diff
@@ -109,6 +109,9 @@ protected static ValueType GetValueType(object value)
             if (value is DynamicDictionary)
                 return ValueType.ConvertToJson;
 
+            if (value is IDictionary && _index.Definition.Version >= IndexDefinitionBaseServerSide.IndexVersion.ProperlyParseDictionaryToStoredField)
```

Comment:
Also need to handle `IDictionary<string,string>`, and a few other permutation, no?

Reply:
https://dotnetfiddle.net/FrjRuL

It works for `IDictionary<string,string>`. I cannot figure out the case it doesn't.

Reply:
Thanks, did *not* know that.